@DQ/Q4.m

```diff
@@ -0,0 +1,58 @@
+% Q4(x) returns the partial derivative of the unit quaternion r with
+%       respect to log(r).
+%       See Eq. (22) of Savino et al (2020). 
+%       Pose consensus based on dual quaternion algebra with application to 
+%       decentralized formation control of mobile manipulators.
+%       https://doi.org/10.1016/j.jfranklin.2019.09.045
+
+% (C) Copyright 2011-2023 DQ Robotics Developers
+% 
+% This file is part of DQ Robotics.
+% 
+%     DQ Robotics is free software: you can redistribute it and/or modify
+%     it under the terms of the GNU Lesser General Public License as published by
+%     the Free Software Foundation, either version 3 of the License, or
+%     (at your option) any later version.
+% 
+%     DQ Robotics is distributed in the hope that it will be useful,
+%     but WITHOUT ANY WARRANTY; without even the implied warranty of
+%     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%     GNU Lesser General Public License for more details.
+% 
+%     You should have received a copy of the GNU Lesser General Public License
+%     along with DQ Robotics.  If not, see <http://www.gnu.org/licenses/>.
+%
+% DQ Robotics website: dqrobotics.github.io
+%
+% Contributors to this file:
+% 1. Bruno Vilhena Adorno (adorno@ieee.org)
+%        - Responsible for the original implementation.
+%
+% 2. Juan Jose Quiroz Omana (juanjqo@g.ecc.u-tokyo.ac.jp)
```

Comment:
@juanjqo Could you add the link to the issue/PR and also mention that you added the check for unit norm?

Reply:
@mmmarinho sure!